linera-execution/src/lib.rs

```diff
@@ -50,6 +51,136 @@ use serde::{Deserialize, Serialize};
 use std::{io, path::Path, str::FromStr, sync::Arc};
 use thiserror::Error;
 
+/// Substracts an amount from a balance and reports an error if that is impossible
+pub fn sub_assign_fees(balance: &mut Amount, fees: Amount) -> Result<(), PricingError> {
+    Ok(balance
+        .try_sub_assign(fees)
+        .map_err(|_| ArithmeticError::Underflow)?)
+}
+
+/// The entries of the runtime related to storage
+#[derive(Copy, Debug, Clone)]
+pub struct RuntimeLimits {
+    /// maximum_budget for reading
+    pub max_budget_num_reads: u64,
+    /// maximum budget of bytes read
+    pub max_budget_bytes_read: u64,
+    /// maximum budget of bytes written
+    pub max_budget_bytes_written: u64,
+    /// The maximum size of read allowed per block
+    pub maximum_bytes_read: u64,
+    /// The maximum size of write allowed per block
+    pub maximum_bytes_written: u64,
+}
+
+/// The entries of the runtime related to storage
+#[derive(Copy, Debug, Clone)]
+pub struct ResourceTracker {
+    /// The used fuel in the computation
+    pub used_fuel: u64,
+    /// The number of reads in the computation
+    pub num_reads: u64,
+    /// The total number of bytes read
+    pub bytes_read: u64,
+    /// The total number of bytes written
+    pub bytes_written: u64,
+    /// The maximum size of read that remains available for use
+    pub maximum_bytes_read: u64,
+    /// The maximum size of write that remains available for use
+    pub maximum_bytes_written: u64,
+}
+
+#[cfg(any(test, feature = "test"))]
+impl Default for ResourceTracker {
+    fn default() -> Self {
+        ResourceTracker {
+            used_fuel: 0,
+            num_reads: 0,
+            bytes_read: 0,
+            bytes_written: 0,
+            maximum_bytes_read: u64::MAX / 2,
+            maximum_bytes_written: u64::MAX / 2,
+        }
+    }
+}
+
+impl Default for RuntimeLimits {
+    fn default() -> Self {
+        RuntimeLimits {
+            max_budget_num_reads: u64::MAX / 2,
+            max_budget_bytes_read: u64::MAX / 2,
+            max_budget_bytes_written: u64::MAX / 2,
+            maximum_bytes_read: u64::MAX / 2,
+            maximum_bytes_written: u64::MAX / 2,
+        }
+    }
+}
+
+impl ResourceTracker {
+    /// Update the limits for the maximum and update the balance.
+    /// The substraction are guaranteed to work correctly and the
```

Comment:
```suggestion
    /// The substractions are guaranteed to work correctly and the
```

(But actually I don't think that needs to be in the method documentation. It would only have to be there if they were _not_ guaranteed to work correctly.)